File: kevoree-corelibrary/javase/org.kevoree.library.javase.mongoDB/src/main/java/org/kevoree/library/javase/mongoDB/Tester.java
package org.kevoree.library.javase.mongoDB;

import com.mongodb.BasicDBObject;
import com.mongodb.DB;
import com.mongodb.DBCollection;
import com.mongodb.Mongo;
import de.flapdoodle.embed.mongo.MongodExecutable;
import de.flapdoodle.embed.mongo.MongodProcess;
import de.flapdoodle.embed.mongo.MongodStarter;
import de.flapdoodle.embed.mongo.config.MongodConfig;
import de.flapdoodle.embed.mongo.distribution.Version;
import de.flapdoodle.embed.process.runtime.Network;

import java.util.Date;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 21/08/12
 * Time: 21:32
 */
public class Tester {

    public static void main(String[] args){

        try {
            int port = 12345;
            MongodProcess mongod = null;
            MongodConfig mongodConfig = new MongodConfig(Version.Main.V2_2, port, Network.localhostIsIPv6());
            MongodStarter runtime = MongodStarter.getDefaultInstance();
            MongodExecutable mongodExecutable = runtime.prepare(mongodConfig);
            mongod = mongodExecutable.start();
            Mongo mongo = new Mongo("localhost", port);
            DB db = mongo.getDB("test");
            DBCollection col = db.createCollection("testCol", new BasicDBObject());
            col.save(new BasicDBObject("testDoc", new Date()));

            Thread.sleep(5000);

            mongod.stop();

        } catch(Exception e){  }

    }

}

File: kevoree-corelibrary/frascati/org.kevoree.library.frascati.mavenplugin/src/main/scala/org/kevoree/library/frascati/mavenplugin/Tester.scala
package org.kevoree.library.frascati.mavenplugin

import java.io.File
import org.kevoree.framework.KevoreeXmiHelper
import org.kevoree.KevoreeFactory
import org.kevoree.impl.DefaultKevoreeFactory

/**
 * Created by IntelliJ IDEA.
 * User: duke
 * Date: 30/01/12
 * Time: 17:44
 */

object Tester extends App {
        println("Hi")
  KevoreeXmiHelper.$instance.save("/tmp/test.kev",CompositeParser.parseCompositeFile(new DefaultKevoreeFactory().createContainerRoot,
new File("/opt/frascati-runtime-1.4/examples/helloworld-pojo/src/main/resources/helloworld-pojo.composite"),"1.5.1-SNAPSHOT","org.kevoree.library.javase","org.kevoree.library.javase.test","helloworld-pojo.composite"))
}